pkg/async/notifications/implementations/gcp_processor.go

```diff
@@ -0,0 +1,91 @@
+package implementations
+
+import (
+	"context"
+	"time"
+
+	"github.com/NYTimes/gizmo/pubsub"
+	"github.com/flyteorg/flyteadmin/pkg/async"
+	"github.com/flyteorg/flyteadmin/pkg/async/notifications/interfaces"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/flyteorg/flytestdlib/promutils"
+	"github.com/golang/protobuf/proto"
+)
+
+// TODO: Add a counter that encompasses the publisher stats grouped by project and domain.
+type GcpProcessor struct {
+	sub           pubsub.Subscriber
+	email         interfaces.Emailer
+	systemMetrics processorSystemMetrics
+}
+
+func NewGcpProcessor(sub pubsub.Subscriber, emailer interfaces.Emailer, scope promutils.Scope) interfaces.Processor {
+	return &GcpProcessor{
+		sub:           sub,
+		email:         emailer,
+		systemMetrics: newProcessorSystemMetrics(scope.NewSubScope("gcp_processor")),
+	}
+}
+
+func (p *GcpProcessor) StartProcessing() {
+	for {
+		logger.Warningf(context.Background(), "Starting GCP notifications processor")
+		err := p.run()
+		logger.Errorf(context.Background(), "error with running GCP processor err: [%v] ", err)
+		time.Sleep(async.RetryDelay)
+	}
+}
+
+func (p *GcpProcessor) run() error {
+	var emailMessage admin.EmailMessage
+	var err error
+
+	for msg := range p.sub.Start() {
+		p.systemMetrics.MessageTotal.Inc()
+
+		if err = proto.Unmarshal(msg.Message(), &emailMessage); err != nil {
```

Comment:
out of curiosity, I know for the AWS implementation we had to do some [massaging](https://github.com/flyteorg/flyteadmin/blob/master/pkg/async/notifications/implementations/processor.go#L64) with the SNS/SQS message format in order to extract the email body from the dequeued subscriber message - is this completely unnecessary with gcp?

Reply:
Yes, it's unnecessary in GCP because its formatting is different from AWS. When I try using AWS' parser on GCP Pubsub, I got this error:

```
{"json":{"src":"processor.go:111"},"level":"debug","msg":"failed to unmarshal to notification object message [\u0012\ufffd\u0001\n\u0026\n\u000bflytesnacks\u0012\u000bdevelopment\"\npgb4a56coo\u0018\u0004\"\u000c\u0008\ufffd\ufffd\ufffd\ufffd\u0006\u0010\ufffdԈ\ufffd\u0001*bgs://flyte-test/metadata/propeller/flytesnacks-development-pgb4a56coo/end-node/data/0/outputs.pb] with err: string field contains invalid UTF-8","ts":"2021-09-14T22:17:27Z"}
```

Reply:
great, good to know that GCP doesn't require any message parsing